source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -225,13 +250,52 @@ class FilterConfigPerRoute : public Router::RouteSpecificFilterConfig {
     return grpc_service_;
   }
 
+  const std::vector<std::string>& untypedForwardingMetadataNamespaces() const {
+    return untyped_forwarding_namespaces_;
+  }
+  const std::vector<std::string>& typedForwardingMetadataNamespaces() const {
+    return typed_forwarding_namespaces_;
+  }
+  const std::vector<std::string>& untypedReceivingMetadataNamespaces() const {
+    return untyped_receiving_namespaces_;
+  }
+
+  enum MergeBehavior {
```

Comment:
For futher PR: I would suggest using proto enum here directly, rather than self-define enum and `initMergeBehavior` converter in the implementation.
This can avoid out of sync situation/breakage if API proto is updated

You can see similar case in ext_proc `ProcessingMode_BodySendMode`

Reply:
This is a good call. I will follow this up in another PR. Thanks for all of the thorough reviews on these PRs!

Reply:
Per offline discussion with @abeyad, it makes more sense to have the merge logic determined by the control plane than to have strong opinions about it here. It will therefore be expected that if metadata options are specified in an override, we will use that entire namespace list, empty or not, and replace the less-specific config with it. If metadata options are not specified in an override on less-specific config, then `absl::nullopt` is used to indicate such, and the namespace vectors will not be affected.